src/libbson/src/bson/bson-error.c

```diff
@@ -134,10 +147,18 @@ bson_strerror_r (int err_code,                    /* IN */
    // POSIX Spec: since strerror_l() is required to return a string for some
    // errors, an application wishing to check for all error situations should
    // set errno to 0, then call strerror_l(), then check errno.
-   errno = 0;
-   ret = strerror_l (err_code, uselocale ((locale_t) 0));
-   if (errno != 0) {
-      ret = NULL;
+   if (locale != (locale_t) 0) {
+      errno = 0;
+      ret = strerror_l (err_code, locale);
+
+      if (errno != 0) {
```

Comment:
I had a fleeting thought of "what if we wanted to translate `errno` here? We can't possibly use `strerror_l()`!"

:laugh-cry: